tests/Unit/Evolution/Systems/NewtonianEuler/Limiters/Test_Flattener.cpp

```diff
@@ -0,0 +1,378 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/NewtonianEuler/Limiters/Flattener.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/IdealFluid.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace {
+
+void test_flattener_1d() noexcept {
+  const Mesh<1> mesh(4, Spectral::Basis::Legendre,
+                     Spectral::Quadrature::GaussLobatto);
+  const EquationsOfState::IdealFluid<false> equation_of_state{5. / 3.};
+  const Scalar<DataVector> det_logical_to_inertial_jacobian(
+      DataVector{{2., 2., 2., 2.}});
+
+  // case with no flattening needed
+  {
+    Scalar<DataVector> mass_density_cons(DataVector{{0.8, 1.6, 2.2, 1.2}});
```

Comment:
Might be good to add `INFO("Check no flattening needed");` instead of the c++ comment, similar below.

These changes are optional, just something to consider :)